torchmultimodal/models/omnivore.py

```diff
@@ -104,11 +110,97 @@ def _omnivore_swin_t_encoder() -> SwinTransformer3d:
     return encoder
 
 
-# TODO: add pretrained weight capability
-def omnivore_swin_t(encoder_only=False) -> nn.Module:
+def _omnivore_swin_s_encoder() -> SwinTransformer3d:
+    encoder = SwinTransformer3d(
+        patch_size=[2, 4, 4],
+        embed_dim=96,
+        depths=[2, 2, 18, 2],
+        num_heads=[3, 6, 12, 24],
+        window_size=[8, 7, 7],
+        stochastic_depth_prob=0.3,
+        norm_layer=nn.LayerNorm,
+        patch_embed=PatchEmbedOmnivore,
+        num_classes=None,
+    )
+    return encoder
+
+
+def _omnivore_swin_b_encoder() -> SwinTransformer3d:
+    encoder = SwinTransformer3d(
+        patch_size=[2, 4, 4],
+        embed_dim=128,
+        depths=[2, 2, 18, 2],
+        num_heads=[4, 8, 16, 32],
+        window_size=[16, 7, 7],
+        stochastic_depth_prob=0.3,
+        norm_layer=nn.LayerNorm,
+        patch_embed=PatchEmbedOmnivore,
+        num_classes=None,
+    )
+    return encoder
+
+
+def omnivore_swin_t(
+    encoder_only: bool = False, pretrained: bool = False, progress: bool = True
+) -> nn.Module:
+    """
+    Builder function to get omnivore model with swin_t variant encoder
+    Args:
+        encoder_only (bool): If true then the builder will only return encoder without head (default: False)
+        pretrained (bool): If true then the it will load pretrained weight,
+            otherwise it will have random weight (default: False)
+        progress (bool): If true then there will be a progress bar for downloading weight (default: True)
+    """
     encoder = _omnivore_swin_t_encoder()
+    heads = _multimodal_head(input_dim=encoder.num_features)
+    model = OmnivoreArchitecture(encoder, heads)
+    if pretrained:
+        model.load_model(_OMNIVORE_PRETRAINED_URLS["swin_t"])
```

Comment:
I think in general we dont really test loading pretrained weight in torchvision now since it is time consuming (the weight is usually large). In fact in some test, we disable the `load_state_dict` function explicitly ([see this function](https://github.com/pytorch/vision/blob/main/test/test_models.py#L34)).

> how do you unit test this? In our case, we create small files to store under test/assets/ and load them locally into the model to test. I've seen torchvision also has asset files, wonder if you consider this a viable approach?

I dont really get what you mean here. I think in general the model file is quite big (several hundred MBs or few GBs). So it is not ideal to store on the repo.